DeployWithLibraries/2_Generate_Metadata/step2.md

```diff
@@ -0,0 +1,32 @@
+# Deploying Library
+
+The **library** from the previous chapter was in the same file as the **contract**. However, they won't be deployed together and will have different addresses.
+
+In order to use a library, the calling contract must have the library's **address**.  
+
+But it doesn't go in the contract's Solidity code.  It goes in the compiled contract's `build artifact` AKA its **metadata**. 
```

Comment:
that's not really clear, also `compiled contract` and `metadata` are 2 different things.
Another way of saying that: 
`The library addresses are not  directly specified in the solidity code. instead of that the compiled bytecode contains placeholder where library addresses should be put it.`

Reply:
is this better:

In order to use a library, the calling contract must have the library's **address**.  

But the library's address is not directly specified in the solidity code. The calling contract's compiled bytecode contains a placeholder where library's addresses will go.

In Remix, we edit the compiled contract's **metadata** - AKA its `build artifact` with the **library's address** and the replacement happens when the **calling contract** is deployed.

Reply:
`we edit the compiled contract's metadata with the library's address and the replacement happens when the calling contract is deployed.`
not really sure that is accurate, can we talk about that?